subsys/bluetooth/host/gatt.c

```diff
@@ -786,28 +908,32 @@ static bool update_range(u16_t *start, u16_t *end, u16_t new_start,
 	return true;
 }
 
-static void sc_indicate(struct gatt_sc *sc, u16_t start, u16_t end)
+static void sc_indicate(struct bt_conn *conn, u16_t start, u16_t end)
```

Comment:
You don't seem to use the conn here? Was that supposed to be passed over to process with gatt_sc.conn?

Reply:
Yes it was supposed to be passed to sc_process using gatt_sc.conn, so to send indication indicate to a specific connected peer or to everyone (conn=NULL). Will fix!

Reply:
Just added the missing assignment in latest push.

Reply:
But the changes usually don't originate from a peer, there either for adding or removing services locally which affects all connections.

Reply:
Yes, but sc_indicate can also be called when a single peer reconnects (see sc_restore function), and in this case we need to send an indication to this specific connection only.

Reply:
@Vudentz When the GATT database is changed then all connected peers are are indicated i.e sc_indicate(conn = NULL). For all bonded devices an update is stored. When that peer connects there will be an indicate to that peer ,i.e sc_indicate(conn = connected_peer) called from sc_restore.

Reply:
I though we wouldn't use sc_indicate for that, or gatt_sc, since in that case we want to indicate immediately and not use a delayed_work for that, gatt_sc aggregate all the changes happening in the meantime if we overwrite it with the range from settings that is no longer a valid range so we would have to lock the gatt_sc in the meantime.

Reply:
Actually this problem already exists in master, since sc_restore can be called multiple times in a row merging the ranges for different peers, we should probably change it to just operate in the connection passed to ccc_update and sending the indication immediately.

Reply:
I just pushed a solution where sc_restore uses bt_gatt_indicate directly instead of sc_indicate. It adds an sc_restore_params array (1 per connection) as well as an sc_restore_rsp callback to handle the change-aware bit in CF.
Seems to work well AFAICanTest.